executor/prepared.go

```diff
@@ -336,7 +336,12 @@ func ResetStmtCtx(ctx context.Context, s ast.StmtNode) {
 	sc.TimeZone = sessVars.GetTimeZone()
 
 	switch stmt := s.(type) {
-	case *ast.UpdateStmt, *ast.DeleteStmt:
+	case *ast.UpdateStmt:
+		sc.IgnoreTruncate = false
+		sc.OverflowAsWarning = false
+		sc.TruncateAsWarning = !sessVars.StrictSQLMode || s.(*ast.UpdateStmt).IgnoreErr
```

Comment:
s/ s.(*ast.UpdateStmt)/ stmt

Reply:
what will happen if `ignore` is used when overflow happens?

Reply:
I wanna to fix the overflow problem in another pr.